File: src/main/java/eu/execom/fabut/enums/AssertType.java
package eu.execom.fabut.enums;

/**
 * Assert type.
 */
public enum AssertType {

    /**
     * Using repository functionality.
     */
    REPOSITORY_ASSERT,

    /**
     * Using regular object assert.
     */
    OBJECT_ASSERT,

    /**
     * When test does not meet necessary prerequisites.
     */
    UNSUPPORTED_ASSERT;
}

File: src/main/java/eu/execom/fabut/enums/AssertType.scala
package eu.execom.fabut.enums

/**
 * Assert type.
 */
class AssertType

object AssertType extends Enumeration {

  type AssertType = Value

  /**
   * Using repository functionality.
   */
  val REPOSITORY_ASSERT = Value

  /**
   * Using regular object assert.
   */
  val OBJECT_ASSERT = Value

  /**
   * When test does not meet necessary prerequisites.
   */
  val UNSUPPORTED_ASSERT = Value
}